https://github.com/NixOS/nix/issues/6046
--- a/src/libutil/util.cc
+++ b/src/libutil/util.cc
@@ -1358,11 +1358,19 @@ std::string shellEscape(const std::string_view s)
 
 void ignoreException()
 {
+  try {
     try {
         throw;
     } catch (std::exception & e) {
         printError("error (ignored): %1%", e.what());
+    } catch(...) {
+        fprintf(stderr, "%s: unknown exception!\n", __func__);
     }
+  } catch (std::exception & e) {
+      fprintf(stderr, "%s: rethown exception: %s\n", __func__, e.what());
+  } catch(...) {
+      fprintf(stderr, "%s: rethown unknown exception!\n", __func__);
+  }
 }
 
 bool shouldANSI()
