diff --git a/src/textview.c b/src/textview.c
index 74ba34d..ae2eb1d 100644
--- a/src/textview.c
+++ b/src/textview.c
@@ -733,7 +733,8 @@ static void textview_add_part(TextView *textview, MimeInfo *mimeinfo)
 			END_TIMING();
 			GTK_EVENTS_FLUSH();
 		}
-	} else if (mimeinfo->type == MIMETYPE_TEXT) {
+	}
+	if (mimeinfo->type == MIMETYPE_TEXT) {
 		if (prefs_common.display_header && (charcount > 0))
 			gtk_text_buffer_insert(buffer, &iter, "\n", 1);
 
