--- a/plugins/actor/G-Force/unix/libmfl/libmfl.h
+++ b/plugins/actor/G-Force/unix/libmfl/libmfl.h
@@ -53,9 +53,9 @@ void mfl_DestroyContext(mfl_context cx);
 void mfl_SetDrawMode(mfl_context cx, int mode);
 
 /* Info */
-inline unsigned int mfl_GetTextWidth(const mfl_context cx, 
+unsigned int mfl_GetTextWidth(const mfl_context cx, 
 				     const char *s);
-inline unsigned int mfl_GetTextWidthL(const mfl_context cx, 
+unsigned int mfl_GetTextWidthL(const mfl_context cx, 
 				      const char *s, int l);
 
 /* Character drawing */
diff --git a/plugins/actor/G-Force/unix/libmfl/mfl.c b/plugins/actor/G-Force/unix/libmfl/mfl.c
index 92af64c..aab0a5b 100644
--- a/plugins/actor/G-Force/unix/libmfl/mfl.c
+++ b/plugins/actor/G-Force/unix/libmfl/mfl.c
@@ -123,12 +123,12 @@ void mfl_SetDrawMode(mfl_context cx, int mode) {
   cx->opmode = mode;
 }
 
-inline unsigned int mfl_GetTextWidthL(const mfl_context cx, 
+unsigned int mfl_GetTextWidthL(const mfl_context cx, 
 				      const char *s, int l) {
   return l * 8;
 }
 
-inline unsigned int mfl_GetTextWidth(const mfl_context cx, const char *s) {
+unsigned int mfl_GetTextWidth(const mfl_context cx, const char *s) {
   return mfl_GetTextWidthL(cx, s, strlen(s));
 }
 
